toolkit/tools/pkg/imagecustomizerlib/resolvconf_test.go

```diff
@@ -0,0 +1,170 @@
+	fakeResolvConfContents := "abcdef"
+	err = file.Write(fakeResolvConfContents, imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = imageConnection.CleanClose()
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Customize image.
+	config = imagecustomizerapi.Config{
+		OS: &imagecustomizerapi.OS{},
+	}
+
+	err = CustomizeImage(buildDir, testDir, &config, outImageFilePath, nil, outImageFilePath, "raw", "",
+		false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	imageConnection, err = connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Ensure resolv.conf file was restored.
+	imageResolvConfPath = filepath.Join(imageConnection.Chroot().RootDir(), resolvConfPath)
+	actualResolvConfContents, err := file.Read(imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	assert.Equal(t, fakeResolvConfContents, actualResolvConfContents)
+
+	// Delete resolv.conf file.
+	err = os.Remove(imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Create a resolv.conf symlink.
+	fakeResolvConfSymlinkPath := "../fake/resolv.conf"
+	err = os.Symlink(fakeResolvConfSymlinkPath, imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = imageConnection.CleanClose()
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Customize image.
+	err = CustomizeImage(buildDir, testDir, &config, outImageFilePath, nil, outImageFilePath, "raw", "",
+		false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	imageConnection, err = connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Ensure resolv.conf symlink was restored.
+	imageResolvConfPath = filepath.Join(imageConnection.Chroot().RootDir(), resolvConfPath)
+	actualResolvConfSymlinkPath, err := os.Readlink(imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	assert.Equal(t, fakeResolvConfSymlinkPath, actualResolvConfSymlinkPath)
+
+	// Delete resolv.conf file.
+	err = os.Remove(imageResolvConfPath)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	err = imageConnection.CleanClose()
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	// Customize image.
+	config = imagecustomizerapi.Config{
+		OS: &imagecustomizerapi.OS{
+			Services: imagecustomizerapi.Services{
+				Enable: []string{"systemd-resolved"},
+			},
+		},
+	}
+
+	err = CustomizeImage(buildDir, testDir, &config, outImageFilePath, nil, outImageFilePath, "raw", "",
+		false /*useBaseImageRpmRepos*/, false /*enableShrinkFilesystems*/)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	imageConnection, err = connectToCoreEfiImage(buildDir, outImageFilePath)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer imageConnection.Close()
+
+	// Ensure resolv.conf symlink was set to systemd-resolved.
```

Comment:
I'd suggest splitting these into separate tests with names indicating the precise scenario being tests. This will make it easier to debug faster which case failed. The code for each scenario will also be shorted, making it simpler to understand and debug.

Reply:
I think it ended up with more code overall. But fixed.

Reply:
Yes, more code overall but less per test. If there's any common boilerplate code, which is common across all tests, I'm also fine with extracting it to a separate function.